pkg/kubelet/dockertools/docker.go

```diff
@@ -150,6 +150,31 @@ func filterHTTPError(err error, image string) error {
 	}
 }
 
+// Check if the inspected image matches what we are looking for
+func matchImageTagOrSHA(inspected dockertypes.ImageInspect, image string) bool {
+	parts := strings.SplitN(image, ":", 2)
+	if len(parts) == 1 {
```

Comment:
Can we reuse the same dockerref code for parsing the given string? https://github.com/kubernetes/kubernetes/blob/9f1c3a4c562a1b5ab96a4c43d95c5c39d80fbec5/pkg/kubelet/dockertools/docker.go#L161